app/controllers/rooms_controller.rb

```diff
@@ -93,7 +93,7 @@ def join
     return redirect_to root_path,
       flash: { alert: I18n.t("administrator.site_settings.authentication.user-info") } if auth_required
 
-    unless @room.owned_by?(current_user)
+    if current_user && !(@room.owned_by?(current_user) || room_shared_with_user)
```

Comment:
You don't need to check for current_user here since .owned_by? and .shared_by? both return false if the current_user is nil